### PR TITLE
Create MOD for "Resistance to All Status Ailments"

### PR DESCRIPTION
**Pull request by neuromancerxi**
**_I affirm:_**
- [x] that I agree to Project Topaz's [Limited Contributor License Agreement](http://project-topaz.com/blob/release/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I've _tested my code_ since the last commit in the PR, and will test after any later commits

Creates MOD 958 - STATUSRES for "Resistance to All Status Ailments".
Removes Death Resistance from Items with "All Status Ailments" (see: related BG Thread with justification - https://www.bluegartr.com/threads/108199-Random-Facts-Thread-Other?s=c66ef360f8f47cb9a568306781e9a045&p=6981981&viewfull=1#post6981981 )
Replaces Individual Resist MODS (240-255) on existing "All Status Ailment" Items with a single 958 with relevant value.
Adds math check in magic.lua to combine specific resistance values with the newly created All Status Ailments MOD.

### CODE REVIEW

File: scripts/globals/status.lua [hunk 1]
```diff
@@ -1557,9 +1558,9 @@ tpz.mod =
 
     -- The spares take care of finding the next ID to use so long as we don't forget to list IDs that have been freed up by refactoring.
     -- 570 - 825 used by WS DMG mods these are not spares.
-    -- SPARE = 958, -- stuff
-    -- SPARE = 959, -- stuff
-    -- SPARE = 960, -- stuff
+    -- SPARE = 964, -- stuff
+    -- SPARE = 965, -- stuff
+    -- SPARE = 966, -- stuff
```
Comment (TeoTwawki) [inline scripts/globals/status.lua:1563]: can you do me the favor of making sure modifier.h has the same "spares" as well?

 the fact you had to jump the 1st spare id all the way to 964 tells me the 2 files have come unsynchornised in their lists again.

Comment (TeoTwawki) [reply scripts/globals/status.lua:1563]: not your fault you just happen to be in prime position to fix it

Comment (neuromancerxi) [reply scripts/globals/status.lua:1563]: https://github.com/neuromancerxi/topaz/blob/003425b91d553a271fd4237119def041b4cf3be0/src/map/modifier.h#L804

modifier.h already has next spare as 964.

Comment (TeoTwawki) [reply scripts/globals/status.lua:1563]: but I looked and it doesn't have 1:1 for all 3.

```cpp
    // The spares take care of finding the next ID to use so long as we don't forget to list IDs that have been freed up by refactoring.
    // 570 through 825 used by WS DMG mods these are not spares.
    // SPARE = 964, // stuff
    // SPARE = 965, // stuff
};
```
both files should have all matching id's, for everything, even the spares. it helps prevent future editors from mistakenly thinking an ID is used/not used

**Comment by kaincenteno** [comment]
is this the MOD for the ribbon equipment? :smile:

**Comment by neuromancerxi** [comment]
> is this the MOD for the ribbon equipment? smile

Yep. Ribbons, Hearty Earring, and others with the "Resistance to All Status Ailments" listed on the item.

File: scripts/globals/magic.lua [hunk 2]
```diff
@@ -447,6 +447,7 @@ end
 -- target has to the given effect (stun, sleep, etc..)
 function getEffectResistance(target, effect)
     local effectres = 0
+    local statusres = target:getMod(tpz.mod.STATUSRES)
     if (effect == tpz.effect.SLEEP_I or effect == tpz.effect.SLEEP_II) then
         effectres = tpz.mod.SLEEPRES
```
Comment (TeoTwawki) [inline scripts/globals/magic.lua:452]: note to self: come back later and murder this giant if/else tower with great fury.

**Review by TeoTwawki — APPROVED** [review]

**Review by 59blargedy — APPROVED** [review]

**Pull request merged**